### PR TITLE
Add ECDSA support to DNSSEC infra via mbedTLS

### PR DESCRIPTION
**Pull request by rgacogne**

### CODE REVIEW

**Comment by ahupowerdns** [comment]
+1

**Comment by Habbie** [comment]
```
mbedtlssigners.cc: In constructor ‘MbedECDSADNSCryptoKeyEngine::MbedECDSADNSCryptoKeyEngine(unsigned int)’:
mbedtlssigners.cc:390:43: error: ‘lexical_cast’ was not declared in this scope
       throw runtime_error("Unknown algo "+lexical_cast<string>(d_algorithm)+" from ECDSA class");
```

possibly as a result of merging #3029 into master today :(

**Comment by rgacogne** [comment]
Rebased, replaced lexical_cast<string>() by std::to_string()

File: pdns/mbedtlscompat.hh [hunk 1]
```diff
@@ -65,10 +69,12 @@ typedef md_type_t mbedtls_md_type_t;
 
 #define mbedtls_entropy_init entropy_init
 #define mbedtls_entropy_func entropy_func
+#define mbedtls_entropy_free entropy_free
 
-#define mbedtls_ctr_drbg_init
+#define mbedtls_ctr_drbg_init ctr_drbg_init
 #define mbedtls_ctr_drbg_seed ctr_drbg_init
```
Comment (mind04) [inline pdns/mbedtlscompat.hh]: Are you sure about this change? mbedtls_ctr_drbg_init() is always flowed by a  mbedtls_ctr_drbg_seed() in the code. And the argument count of mbedtls_ctr_drbg_init and ctr_drbg_init do not match

Comment (rgacogne) [inline pdns/mbedtlscompat.hh]: You are right, I have been fooled by the include/mbedtls/compat-1.3.h file shipped with mbedtls sources, which contains:
```
#define ctr_drbg_init mbedtls_ctr_drbg_init
```

I will revert that, thank you!

**Comment by rgacogne** [comment]
Reverted the mbedtls_ctr_drbg_init define in pdns/mbedtlscompat.hh

**Comment by Habbie** [comment]
Travis was unhappy for remotebackend reasons, have restarted the builds. If still red in the morning, rebase suggested because I don't know.

**Comment by rgacogne** [comment]
Rebased, still fails on remotebackend-pipe-dnssec, remotebackend-unix-dnssec, remotebackend-http-dnssec and remotebackend-zeromq-dnssec. Any idea where I should look?

**Comment by Habbie** [comment]
I'll have a go at it. Will report back :)

**Comment by Habbie** [comment]
('works on my machine' was WITH botan, now testing without)

**Comment by Habbie** [comment]
No longer works on my machine.

**Comment by rgacogne** [comment]
Fix SHA384 hash returning a 64-byte string instead of an 48-byte one. Fix white-spaces issues.

**Pull request merged**